sympy/polys/fields.py

```diff
@@ -237,10 +237,11 @@ def _rebuild_expr(self, expr, mapping):
             if gen.is_Pow or isinstance(gen, ExpBase))
 
         def _rebuild(expr):
-            generator = mapping.get(expr)
 
-            if generator is not None:
-                return generator
+            if mapping.get(expr) is not None:
+                return mapping.get(expr)
+            elif mapping.get(1/expr) is not None:
+                return 1/mapping.get(1/expr)
```

Comment:
Shouldn't this be handled by the `is_Pow` check below?

Reply:
No right !? Something like the following (1st one) would be handled by the `is_Pow `check and this works fine but any containment check for `x` in `QQ.frac_field(1/x)` is failing because there isn't anything I can see handling these checks currently
```
>>> ((x**2) in QQ.frac_field(1/x))
True

>>> ((x**2 + x) in QQ.frac_field(1/x))
False

>>> (x in QQ.frac_field(1/x))
False
```

Reply:
Maybe `QQ.frac_field(1/x)` should just canonicalise to `QQ.frac_field(x)`.

Reply:
Ohh ok , so do you mean to say we should have something like the following
```
>>> QQ.frac_field(1/x)
QQ(x)
>>> QQ.frac_field(1/x**2)
QQ(x**2)
```
Even this would work and seems correct too in all aspects ! Do you have any doubts regarding the mathematical correctness here ?? I ran some checks for cases involving higher negative powers like (`1/x**2` etc) and also multivariate cases. Everything what can be achieved from the suggestion is being achieved by the committed 2 lines of code ! We can surely go for the change if we have no doubt about the mathematical correctness

Reply:
I've added Tests to support the argument made above in the latest commit